executor/linux/build.go

```diff
@@ -520,6 +535,66 @@ func (c *client) ExecBuild(ctx context.Context) error {
 			continue
 		}
 
+		// iterate through each secret provided in the pipeline
+		for _, secret := range c.pipeline.Secrets {
+			_log, err := step.LoadLogs(c.init, &c.stepLogs)
+			if err != nil {
+				return err
+			}
+
+			_log.SetData([]byte(""))
+
+			// ignore pulling secrets coming from plugins
+			if !secret.Origin.Empty() {
+				continue
+			}
+
+			c.Logger.Infof("pulling secret %s", secret.Name)
```

Comment:
thoughts on moving this `Infof` down below the `continue`? the log makes me think its pulling everything